openwisp_monitoring/device/tests/__init__.py

```diff
@@ -62,32 +69,40 @@ def create_test_adata(self):
             m = Metric.objects.get(key=ifname, field_name='clients', object_id=d.pk)
             points = m.read(limit=10, order='time DESC')
             self.assertEqual(len(points), len(iface['wireless']['clients']))
+        return dd
 
-    def _create_multiple_measurements(self, create=True):
+    def _create_multiple_measurements(self, create=True, no_resources=False, count=4):
```

Comment:
It would be better to have this kind of refactors in a different PR to make the reviewing process easier

Reply:
If it wouldn't had increased the speed, I would have desisted from making the change. I have compared the speeds previously with and without `resources` `metrics`, `charts`, `alertsettings` being created.
Increase in individual test's speed was slightly more than 0.1s. If `0.3s` is the threshold then `0.1s` counts according to me. Anyways, for the record, the fastest travis build when there was no changes in `device` tests was slightly less than **29s**. So, I think reducing requests is increasing speed by approximately **5s** overall.
Also, this parameter is only being used when there is no need for resources metrics at all in the test. For example, `test_200_traffic_counter_incremented`, `test_200_traffic_counter_reset`, etc.

If you want to compare the speed increase in slow tests:~
Speed log before changes : [travis-build](https://travis-ci.com/github/nepython/openwisp-monitoring/jobs/348784253#L1072)
Speed log after changes : [travis-build](https://travis-ci.org/github/openwisp/openwisp-monitoring/jobs/699070313#L1071)